hera_pspec/pspecdata.py

```diff
@@ -91,13 +96,36 @@ def validate_datasets(self):
         Validate stored datasets and weights to make sure they are consistent
         with one another (e.g. have the same shape, baselines etc.).
         """
-        # Sanity checks on input data
-        assert len(self.dsets) > 1
-        assert len(self.dsets) == len(self.wgts)
+        # check dsets and wgts have same number of elements
+        if len(self.dsets) != len(self.wgts):
+            raise ValueError("self.wgts does not have same length as self.dsets")
 
-        # Check if data are all the same shape
+        # Check if dsets are all the same shape along freq axis
         Nfreqs = [d.Nfreqs for d in self.dsets]
-        assert np.unique(Nfreqs).size == 1
+        if np.unique(Nfreqs).size > 1:
+            raise ValueError("all dsets must have the same Nfreqs")
+
+        # Check shape along time axis
+        Ntimes = [d.Ntimes for d in self.dsets]
+        if np.unique(Ntimes).size > 1:
+            raise ValueError("all dsets must have the same Ntimes")
+
+        # raise warnings if times don't match
+        if self.warn:
+            lst_diffs = np.array(map(lambda dset: np.unique(self.dsets[0].lst_array) - np.unique(dset.lst_array), self.dsets[1:]))
+            if np.max(np.abs(lst_diffs)) > 0.001:
+                print("Warning: taking power spectra between LST bins misaligned by more than 15 seconds")
+
+        # raise warning if frequencies don't match       
+        if self.warn:
+            freq_diffs = np.array(map(lambda dset: np.unique(self.dsets[0].freq_array) - np.unique(dset.freq_array), self.dsets[1:]))
+            if np.max(np.abs(lst_diffs)) > 0.1e6:
+                print("Warning: taking power spectra between frequency bins misaligned by more than 0.1 MHz")
+
+        # Check for the same polarizations
+        pols = set(map(lambda dset: tuple(sorted(dset.polarization_array)), self.dsets))
+        if np.unique(pols).size > 1:
+            raise ValueError("all dsets must have the same number and kind of polarizations")
```

Comment:
Can you include the polarisations that were detected in each dataset in the error string please? This will help users to debug.